File: backend/update_admin_password.py
import mysql.connector
import os
from dotenv import load_dotenv

load_dotenv()

# Database configuration
MYSQL_CONFIG = {
    'host': os.getenv('DB_HOST', 'localhost'),
    'user': os.getenv('DB_USER', 'root'),
    'password': os.getenv('DB_PASSWORD', ''),
    'database': os.getenv('DB_NAME', 'coreq_loans')
}

# Connect to MySQL
mysql_conn = mysql.connector.connect(**MYSQL_CONFIG)
cursor = mysql_conn.cursor()

# Fresh bcrypt hash for password "1234"
hashed_password = '$2b$08$jEeeu3d0DytLEQARmO4uVO27ulxeJ//l7IIwQE02/kRK4TQRJGSp6'

# Update the user password
cursor.execute(
    "UPDATE users SET password = %s WHERE email = %s",
    (hashed_password, 'admin@coreqcapital.com')
)
mysql_conn.commit()

# Verify
cursor.execute("SELECT id, name, email FROM users WHERE email = %s", ('admin@coreqcapital.com',))
user = cursor.fetchone()
print(f"Updated user:")
print(f"  ID: {user[0]}")
print(f"  Name: {user[1]}")
print(f"  Email: {user[2]}")

cursor.close()
mysql_conn.close()
print("\nPassword updated successfully!")
print("You can now login with:")
print("  Email: admin@coreqcapital.com")
print("  Password: 1234")

File: backend/update-admin-password.js
const bcrypt = require('bcryptjs');
const mysql = require('mysql2/promise');

async function updateAdminPassword() {
  let connection;

  try {
    // Get database credentials from environment
    const { DB_HOST, DB_USER, DB_PASSWORD, DB_NAME, DB_PORT } = process.env;

    if (!DB_HOST || !DB_USER || !DB_PASSWORD || !DB_NAME) {
      console.error('❌ Database credentials not found in environment variables');
      console.error('Available env vars:', Object.keys(process.env).filter(k => k.includes('DB') || k.includes('MYSQL')));
      process.exit(1);
    }

    console.log('Connecting to database...');
    console.log('Host:', DB_HOST);
    console.log('Database:', DB_NAME);

    connection = await mysql.createConnection({
      host: DB_HOST,
      port: DB_PORT || 3306,
      user: DB_USER,
      password: DB_PASSWORD,
      database: DB_NAME
    });

    console.log('✓ Connected to database');

    // New password
    const newPassword = 'Admin@5432';
    console.log('\nGenerating password hash...');
    const hashedPassword = await bcrypt.hash(newPassword, 10);

    // Update password
    console.log('Updating admin password...');
    const [result] = await connection.query(
      'UPDATE users SET password = ? WHERE email = ?',
      [hashedPassword, 'admin@coreqcapital.com']
    );

    if (result.affectedRows > 0) {
      console.log('\n✅ Admin password updated successfully!');
      console.log('\nNew Credentials:');
      console.log('  Email: admin@coreqcapital.com');
      console.log('  Password: Admin@5432');
    } else {
      console.log('\n⚠️  No admin user found with email: admin@coreqcapital.com');
    }

  } catch (error) {
    console.error('\n❌ Error:', error.message);
    process.exit(1);
  } finally {
    if (connection) {
      await connection.end();
    }
  }
}

updateAdminPassword();
